File: token-validation-core/src/main/java/no/nav/security/token/support/core/exceptions/JwtTokenValidatorException.java
package no.nav.security.token.support.core.exceptions;

import java.util.Date;

public class JwtTokenValidatorException extends RuntimeException {

    private final Date expiryDate;

    public JwtTokenValidatorException(String msg) {
        this(msg, null, null);
    }

    public JwtTokenValidatorException(String msg, Throwable cause) {
        this(msg, null, cause);
    }

    public JwtTokenValidatorException(String msg, Date expiryDate, Throwable cause) {
        super(msg, cause);
        this.expiryDate = expiryDate;
    }

    public Date getExpiryDate() {
        return expiryDate;
    }
}

File: token-validation-core/src/main/kotlin/no/nav/security/token/support/core/exceptions/JwtTokenValidatorException.kt
package no.nav.security.token.support.core.exceptions

import java.util.*

class JwtTokenValidatorException @JvmOverloads constructor(msg : String? = null, val expiryDate : Date? = null, cause : Throwable? = null) : RuntimeException(msg, cause) {
    constructor(msg : String, cause : Throwable) : this(msg, null,cause)
}